truss-chains/truss_chains/deployment/code_gen.py

```diff
@@ -725,17 +724,18 @@ def gen_truss_chainlet(
             display_name=dep.display_name,
             options=dep.options,
         )
+    gen_root = pathlib.Path(tempfile.gettempdir())
     chainlet_dir = _make_chainlet_dir(chain_name, chainlet_descriptor, gen_root)
     logging.info(
         f"Code generation for Chainlet `{chainlet_descriptor.name}` "
         f"in `{chainlet_dir}`."
     )
-    _make_truss_config(
-        chainlet_dir,
-        chainlet_descriptor.chainlet_cls.remote_config,
-        dep_services,
-        model_name,
-        use_local_chains_src,
+    write_truss_config_yaml(
+        chainlet_dir=chainlet_dir,
+        chains_config=chainlet_descriptor.chainlet_cls.remote_config,
+        model_name=model_name or chain_name,
```

Comment:
Chain models have their name [suffixed](https://github.com/basetenlabs/truss/blob/76468b37b4715cfb041e69fee68a4557a44e094e/truss-chains/truss_chains/deployment/deployment_client.py?plain=1#L170-L171), but for traditional trusses we'd want to preserve the name specified in `RemoteConfig` exactly (or the one autoderived from the class name)

We can push this coalesce down into `write_truss_config_yaml` if we think that's cleaner